hushline/model.py

```diff
@@ -297,3 +307,106 @@ def __init__(self) -> None:
 
     def __repr__(self) -> str:
         return f"<InviteCode {self.code}>"
+
+
+# Paid tiers
+class Tier(Model):
+    __tablename__ = "tiers"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(db.String(255), unique=True)
+    monthly_amount: Mapped[int] = mapped_column(db.Integer)  # in cents USD
+    stripe_product_id = mapped_column(db.String(255), unique=True)
+    stripe_price_id = mapped_column(db.String(255), unique=True)
+
+    def __init__(self, name: str, monthly_amount: int) -> None:
+        super().__init__()
+        self.name = name
+        self.monthly_amount = monthly_amount
+
+
+class StripeEvent(Model):
+    __tablename__ = "stripe_events"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    event_id: Mapped[str] = mapped_column(db.String(255), unique=True, index=True)
+    event_type: Mapped[str] = mapped_column(db.String(255))
+    event_data: Mapped[str] = mapped_column(db.Text)
+    created_at: Mapped[datetime] = mapped_column(default=datetime.now)
+    status: Mapped[str] = mapped_column(db.String(255), default="pending")
+
+    def __init__(self, event: Event) -> None:
+        super().__init__()
+        self.event_id = event.id
+        self.event_type = event.type
+        self.event_data = str(event)
+
+
+class StripeInvoiceStatusEnum(enum.Enum):
+    DRAFT = "draft"
+    OPEN = "open"
+    PAID = "paid"
+    UNCOLLECTIBLE = "uncollectible"
+    VOID = "void"
+
+
+class StripeInvoice(Model):
+    __tablename__ = "stripe_invoices"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    customer_id: Mapped[str] = mapped_column(db.String(255))
+    invoice_id: Mapped[str] = mapped_column(db.String(255), unique=True, index=True)
+    hosted_invoice_url: Mapped[str] = mapped_column(db.String(255))
```

Comment:
How long are these URLs, should we up the limit to 2048 (the practical URL limit)?

Reply:
They are pretty short, but who knows maybe someday they'll get longer so I'm good with 2048 chars

Reply:
Fixed in https://github.com/scidsg/hushline/pull/595/commits/c627d608ea751a5b26ce6795c2e9149f5e42defb